spec/features/remember_device/cookie_expiration_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe 'signing in with remember device and closing browser' do
+  include SamlAuthHelper
+
+  let(:cookies) do
+    Capybara.
+      current_session.
+      driver.
+      browser.
+      current_session.
+      instance_variable_get(:@rack_mock_session).
+      cookie_jar.
+      instance_variable_get(:@cookies)
+  end
+
+  it 'expires the remember device cookie' do
+    # simulate a user signing in, checking the "remember this browser" checkbox,
+    # and then closing their browser. They should be prompted for MFA again.
+    user = user_with_2fa
+    sign_in_user(user)
+    check t('forms.messages.remember_device')
+    fill_in_code_with_last_phone_otp
+    click_submit_default
+    first(:link, t('links.sign_out')).click
+
+    # expire all cookies, see http://jamesferg.com/testing/bdd/hacking-capybara-cookies/
+    cookies.reject! do |existing_cookie|
+      existing_cookie.expired? != false
+    end
```

Comment:
Would it be clearer if we inverted this?
```suggestion
    cookies.select! do |existing_cookie|
      !existing_cookie.expired?
    end
```

Reply:
Trying again
```suggestion
    cookies.reject!(&:expired?)
```

Reply:
hm it is breaking the test since `#expired?` returns `nil` for session cookies. We only want to keep cookies if that method returns `false`. Going to leave it as-is (although will pass an inline block since it's short)